File: e114/python/solution.py
class Solution:
    """
    @param n and m: positive integer(1 <= n , m <= 100)
    @return an integer
    """ 
    def uniquePaths(self, m, n):
        # write your code here
        ret = 1
        for i in range(n, m+n-1):
            ret *= i
        for j in range(1, m):
            ret /= j
        return ret

File: e114/java/Solution.java
import java.math.BigInteger;
public class Solution {
    /**
     * @param n, m: positive integer (1 <= n ,m <= 100)
     * @return an integer
     */
    public int uniquePaths(int m, int n) {
        // write your code here 
        int more = m > n?m:n;
        int less = m < n?m:n;
        
        BigInteger ret = new BigInteger("1");
        for (Integer i=more; i<=less+more-2; i++) {
            ret = ret.multiply(new BigInteger(i.toString()));
        }
        for (Integer j=1; j<=less-1; j++) {
            ret = ret.divide(new BigInteger(j.toString()));
        }
        return ret.intValue();
    }
}
